shopfloor/services/zone_picking.py

```diff
@@ -647,15 +656,30 @@ def set_destination(
         move_line = self.env["stock.move.line"].browse(move_line_id)
         if not move_line.exists():
             return self._response_for_start(message=self.msg_store.record_not_found())
+
+        pkg_moved = False
         search = self.actions_for("search")
-        # When the barcode is a location
-        location = search.location_from_scan(barcode)
-        if location:
-            response = self._set_destination_location(
-                zone_location, picking_type, move_line, quantity, confirmation, location
-            )
-            if response:
-                return response
+        accept_only_package = not self._move_line_full_qty(move_line, quantity)
+        if not accept_only_package:
+            # When the barcode is a location
+            location = search.location_from_scan(barcode)
+            if location:
+                response = self._set_destination_location(
+                    zone_location,
+                    picking_type,
+                    move_line,
+                    quantity,
+                    confirmation,
+                    location,
+                )
+                if response:
+                    return response
+                else:
+                    # TODO we should have a better way to determine this.
+                    # Right now is based on the fact that set destination
+                    # returns a response if we have to redirect somewhere else
+                    # because of issues.
+                    pkg_moved = True
```

Comment:
It is wrong here, `_set_destination_location` and `_set_destination_package` return also a response for the `zero_check` case (which is not an error, just the normal next step for the operator if zero check is enabled).
So yes we should find another way to know that the package has been moved, I propose:

- for `_set_destination_location`, check if `move_line.qty_done == quantity and move_line.location_dest_id == location`
- for `_set_destination_package`, check if `move_line.qty_done == quantity and move_line.result_package_id == package and move_line.shopfloor_user_id == self.env.uid`

Reply:
makes sense, I'll check

Reply:
@sebalix fixed here https://github.com/camptocamp/wms/pull/42/commits/e27a9a527ebd38ba68549bdee62ec2ebd6961a36